activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -410,7 +414,7 @@ def collation
       def tables(name = nil, database = nil, like = nil) #:nodoc:
         sql = "SHOW TABLES "
         sql << "IN #{quote_table_name(database)} " if database
-        sql << "LIKE #{quote(like)}" if like
+        sql << "LIKE #{quote_like(like)}" if like
```

Comment:
I think we can use `https://github.com/rails/rails/blob/d5902c9e7eaba4db4e79c464d623a7d7e6e2d0e3/activerecord/lib/active_record/sanitization.rb#L115` right?